pkg/registry/apps/statefulset/strategy.go

```diff
@@ -103,7 +106,16 @@ func (statefulSetStrategy) PrepareForUpdate(ctx context.Context, obj, old runtim
 	if !apiequality.Semantic.DeepEqual(oldStatefulSet.Spec, newStatefulSet.Spec) {
 		newStatefulSet.Generation = oldStatefulSet.Generation + 1
 	}
+}
 
+// dropStatefulSetDisabledFields drops fields that are not used if their associated feature gates
+// are not enabled.
+func dropStatefulSetDisabledFields(newSS *apps.StatefulSet, oldSS *apps.StatefulSet) {
+	if !utilfeature.DefaultFeatureGate.Enabled(features.StatefulSetMinReadySeconds) {
+		if oldSS == nil {
+			newSS.Spec.MinReadySeconds = int32(0)
+		}
```

Comment:
This is wrong, you want to always set this field to 0 when the feature is off, so:
```golang
if newSS != nil {
    newSS.Spec.MinReadySeconds = int32(0)
}
```

Reply:
We should be using the oldSS only, I slightly tweaked the defaulting. Now, if the minReadySeconds is >0 then only old value will be copied to new value if not the defaulting to 0 happens.

Reply:
To be clear, following scenarios happen
- When the user disables feature flag and gives a -ve value for minReadySeconds, it will be defaulted to 0.
- When the user enables this flag and gives a -ve value for minReadySeconds, the validation will fail and the user will not be able to create a StatefulSet

This ensures updates go smoothly.